error_formatter.go

```diff
@@ -68,7 +68,7 @@ func (ef *termErrorFormatter) Format(err error) string {
 	switch err := err.(type) {
 	case RuntimeError:
 		return ef.formatRuntime(&err)
-	case parser.StaticError:
+	case errors.StaticError:
```

Comment:
Something to consider: Make `errors.StaticError` unexported, and have `errors.MakeStaticError` (possibly renamed to just `errors.Static`) return an `error`. Provide a function in the `errors` package like so:
```go
func SourceLocation(err error) (ast.LocationRange, bool)
```
In that function, if `err` is of type `*StaticError`, return `(err.(*StaticError)).Loc()` and `true`. You can use that here to avoid sniffing for specific error types, and could instead dispatch on what kind of information is associate with these (opaque) errors.

Reply:
Hmmm... I don't see anything bad about distinguishing between error types (especially when the division is so clean as here - RuntimeError, StaticError and everything else being considered an Internal Error). IIRC the original reason for MakeStaticError was not to hide anything, but to make it harder to forget about location.

Reply:
That's fine. I just saw an opportunity to shrink the surface area of the `errors` package. Since it's internal, we don't want to make its types and functions easy to leak any further.